src/GitTfs/Core/GitRepository.cs

```diff
@@ -808,7 +808,9 @@ public void UseGitIgnore(string pathToGitIgnoreFile)
                 int changesetId;
                 if (TryParseChangesetId(c.Message, out changesetId))
                 {
-                    pairs.Add(changesetId, c.Sha);
+                    if (pairs.TryGetValue(changesetId, out var commitSha))
+                        Trace.TraceWarning("warning: Git commit {0} couldn't be assigned to TFS changeset '{1}' as git commit '{2}' was assigned already. Please correct the export file accordingly if needed", c.Sha, changesetId, commitSha);
+                    else pairs.Add(changesetId, c.Sha);
```

Comment:
```suggestion
                    else
                        pairs.Add(changesetId, c.Sha);
```

Reply:
Overall, i like the change. Maybe you can incorporate the minor style issue and then this can be merged.

Sorry for the long delay, I am currently quite busy with other topics.